src/openlcb/CDIUtils.hxx

```diff
@@ -118,7 +118,64 @@ public:
         return n->text;
     }
 
-    /// Clears out al luserinfo structure pointers. This is necessary to use
+    /// Finds the min value of the xml integer, if there isn't one it returns
+    /// a null pointer
+    /// @param "node" is a CDI element (segment, group, or config element).
+    /// @param "found" will be set to true if there was a minimum found, false if 
+    /// the XML did not contain the respective tag.
+    /// @return contents of the <min> tag, unspecified if !found. 
+    static int find_node_min(const XMLNode *node, bool* found)
+    {
+        auto *n = find_child_or_null(node, "min");
+        // if there is no min defined set bool to false, return any value
+        if (n == nullptr || n->text == nullptr){
+            *found = false;
+            return 0;//there is no defined min
+        }
+        // if the min was found set the bool to true and return the min
+        *found = true;
+        return atoi(n->text); //convert to int and return min value
+    }
+
+    /// Finds the max value of the xml integer, if there isn't one it returns
+    /// a null pointer
+    /// @param "node" is a CDI element (segment, group, or config element).
+    /// @param "found" will be set to true if there was a maximum found, false if 
+    /// the XML did not contain the respective tag.
+    /// @return contents of the <max> tag, unspecified if !found. 
+    static int find_node_max(const XMLNode *node, bool* found)
+    {
+        auto *n = find_child_or_null(node, "max");
```

Comment:
we should try to avoid copy-pasting code like this, because it makes it super hard to fix bugs later.

This should be refactored to use a shared helper function, like

static int find_numeric_element(const XMLNode* parent, const char* tag, bool* found) {
  // move implementation here, use `tag` instead of "min" "max" "default"
}

static int find_node_max(const XMLNode *node, bool* found) {
  return find_numeric_element(node, "max", found);
}

Reply:
modified, and tested manually, also merged other branch with validation testing and that tested good as well.